### ISSUE TITLE
Reformat Review Tab

### ISSUE THREAD

**Issue by happivize** [happivize/live#14]
The review tab out of the box from WooCommerce Product Reviews Pro is too spreadout. I would like to use the following website as a template, with a few tweaks. This is from:

https://www.boombycindyjoseph.com/products/boomstick-color

![image](https://user-images.githubusercontent.com/28787692/34894201-71336520-f7ae-11e7-9cd7-3b8c7c34fc90.png)
https://www.screencast.com/t/IOmqP7Rbd7T

Here's how the write a review should look

https://www.screencast.com/t/d5LWHUWfb7n7

**Comment by happivize** [comment]
some more images:

![image](https://user-images.githubusercontent.com/28787692/34895461-e3df4d1e-f7b3-11e7-8384-298976350965.png)
 
https://www.screencast.com/t/3MoIsncvJ3Cq

**Comment by happivize** [comment]
Thanks. It's close.

1) Bars should be pink, not purple and starts should be orange (on the leave a review)

2) It doesn't look like this line was addressed: 

Here's how the write a review should look

https://www.screencast.com/t/d5LWHUWfb7n7

3) There are avatars, most people do not have an avatar, can they show if they have one and if they don't have one, not show?

**Comment by shpp-shablyuk** [comment]
Check please

**Comment by shpp-shablyuk** [comment]
[style.txt](https://github.com/happivize/live/files/1661487/style.txt)
this to "flatsome-child/style/css"

[review.zip](https://github.com/happivize/live/files/1661490/review.zip)
this to "flatsome-child/woocommerce/single-product" like https://prnt.sc/i53hc5

**Comment by happivize** [comment]
this didn't work.

Here are the files I added per your screenshot:

![image](https://user-images.githubusercontent.com/28787692/35421844-7065bda4-0213-11e8-8c66-b8c60b5d4bbe.png)

and then i added all of the text in that style.txt to the style.css here is the file i downloaded after i added it

https://www.dropbox.com/s/6xgpkyxku59cnx6/style%20from%20live%20012518.txt.css?dl=0

**Comment by shpp-shablyuk** [comment]
I see that the file structure on the live site is not correct. You can go to the dev site 

> wp-content/themes/flatsome-child/woocommerce/single-product

 and look for an example of what kind of folder and files structure there is

**Comment by happivize** [comment]
I am confused, it looks to me in the screenshot you sent that you have the
new files in Contributions folder,do you not have them in contributions?
They are showing right under Contributions. What is the path they should be
in?

**Comment by shpp-shablyuk** [comment]
> wp-content/themes/flatsome-child/woocommerce/single-product

in "single-product" this https://prnt.sc/i64tus

> wp-content/themes/flatsome-child/woocommerce/single-product/contributions

in "single-product/contributions" this https://prnt.sc/i64ujw

**Comment by happivize** [comment]
OK,still some bugs. I  made a video here:

https://www.screencast.com/t/Oq8VQBoiBBX

Also when there are no reviews, it's not possible to leave a review:


[image: Inline image 3]



Here's the file structure I have now:

[image: Inline image 1]

**Comment by shpp-shablyuk** [comment]
I don't see images. Tomorrow I'll look what the problem is

**Comment by happivize** [comment]
OK I forwarded the email with images to bulld0zer@yahoo.com

**Comment by shpp-shablyuk** [comment]
> Also when there are no reviews, it's not possible to leave a review

I fixed this on the dev site

**Comment by shpp-shablyuk** [comment]
These buttons https://prnt.sc/i6q51x are added here https://prnt.sc/i6pesn

**Comment by shpp-shablyuk** [comment]
> Here's the file structure I have now

Yes, now everything is correct.

**Comment by happivize** [comment]
I do not want those extra two buttons up there, they should just have two
choices the review and the question, just should have the two as per the
reference site:

https://www.screencast.com/t/IOmqP7Rbd7T

The post a video and photo should be allowed once they start the process of
the review or question.

**Comment by shpp-shablyuk** [comment]
You can remove them in the admin wordpress, where I showed: https://prnt.sc/i6pesn

**Comment by happivize** [comment]
But I want them to be able to add a video or photo to the review, just not
up at the top

**Comment by shpp-shablyuk** [comment]
Now on the dev site, the buttons for adding photo and video work the way you want?

**Comment by happivize** [comment]
Yes, I just tested adding a photo and it worked.

**Comment by shpp-shablyuk** [comment]
If you specify the settings as in the screenshot https://prnt.sc/i6pesn, the buttons above will be deleted. And adding photos and videos under the review will remain

**Comment by happivize** [comment]
OK, what is updated on dev to make leaving a first review is possible?

**Comment by shpp-shablyuk** [comment]
You should replace the file https://prnt.sc/i6qw8n with this one [contributions.php.zip](https://github.com/happivize/live/files/1671227/contributions.php.zip)

**Comment by happivize** [comment]
That is a zip file but i just see the one file, is there more than one? I
had renamed the old files to have the word PAUSE in them, delte the ond
contributions.zip file and replaced it with this one, but I just got a
bunch of gobbledy gook. Can you send me the files separate instead of a
zip? You can email them to me at eftcolumbus@gmail.com if you can't send
them singly through github.

**Comment by happivize** [comment]
Ok, the latest file you sent when i uploaded it, everything finally worked. So I am closing this out. Thanks.

**Comment by happivize** [comment]
I had someone leave a question on a product, but there is no option to Reply anymore - I used to have it it so that I or the product author could respond to a question or a review.

BTW I had a fresh copy of live put on to dev

**Comment by shpp-shablyuk** [comment]
I can't find this option. Where was it?

**Comment by happivize** [comment]
It would give you the option to reply under a review.

Here is an example of a product where the author, Chameice, responded to a
comment, go to reviews tab

https://happivize.com/product/health-wealth-reset-chameice-daniel/

BTW the contents tab is higher priority over this.

**Comment by shpp-shablyuk** [comment]
I tried to disable the plugin Woo Review Pro, enable another theme, but I don’t see the REPLY button

**Comment by happivize** [comment]
I believe it is a function of Review Pro, so you would have to disable the
other changes you made to that plugin to see original functionality

**Comment by shpp-shablyuk** [comment]
https://prnt.sc/iaxjut
I added a form for commenting on the dev site

**Comment by shpp-shablyuk** [comment]
It is always open.

**Comment by shpp-shablyuk** [comment]
To correct the number of reviews, you need to add a fake comment and then delete it. After that, the number should be equal

**Comment by happivize** [comment]
I don't want it always open, only when clicked

**Comment by shpp-shablyuk** [comment]
I can add a button for this. Where should I place it?

**Comment by happivize** [comment]
![image](https://user-images.githubusercontent.com/28787692/35867653-4b1808ea-0b28-11e8-9a47-4399df8cdfd3.png)

**Comment by shpp-shablyuk** [comment]
Done.
Check please.
[health-wealth-reset-chameice-daniel](https://dev.happivize.com/product/health-wealth-reset-chameice-daniel/#tab-reviews)

**Comment by happivize** [comment]
Looks good. I hope it's a simple add to live? Last time this was not
straightforward.

**Comment by shpp-shablyuk** [comment]
snippet - https://prnt.sc/ibdzva
I sent the file "action.php"
Put it here:
"flatsome-child/woocommerce/single-product/contributions/actions.php"

**Comment by happivize** [comment]
this didn't work, here is video showing both on IE and Foxfire and purging
cache.

https://www.screencast.com/t/ZfDZGZbb7a

i added the file here:

http://50.31.114.157/public/wp-content/themes/flatsome-child/woocommerce/single-product/contributions/actions.php

and i have activated this snippet

https://www.screencast.com/t/lR6pkuS8V53

[image: Inline image 1]

**Comment by shpp-shablyuk** [comment]
sorry, Iforgot this
> You should add comments in the Woo settings: https://prnt.sc/ibflfb

**Comment by shpp-shablyuk** [comment]
========================

>To correct the number of reviews, you need to add a fake comment and then delete it. After that, the number should be equal

**Comment by happivize** [comment]
OK, that helped, but it does not allow for replies to questions, which is
the reason I noticed the missing reply in the first place, see these two:

https://www.screencast.com/t/HdvdQNwQLEP

It's
https://happivize.com/product/hair-skin-and-nails-immortal-beauty-rann-goldrich/


[image: Inline image 1]

**Comment by shpp-shablyuk** [comment]
Fixed
https://dev.happivize.com/product/hair-skin-and-nails-immortal-beauty-rann-goldrich/#tab-reviews

**Comment by happivize** [comment]
ok i hope it's just updating the snippet?

**Comment by shpp-shablyuk** [comment]
No, You should replace "flatsome-child/woocommerce/single-product/contributions/actions.php"
I sent it to email

**Comment by happivize** [comment]
Thanks, it is working now.

**Comment by happivize** [comment]
I have just updated all the plugins on dev and the reformatted reviews aren't working as we had it. I don't know if it's woo 3.3 or new wordpress or new flatsome theme. Can you fix to have it back where we were?

I did not buy the new product reviews pro plugin because their customer support has been uselss, but I did find a free version of it on https://gpldl.com/repository/premium-woocommerce-extensions/ and downloaded it here

https://www.dropbox.com/sh/qr0bd2y3y1lylqi/AAB2al3W7dhal4LJCdPJs-C7a?dl=0

I am just not sure if their version is free of any bad code.

Can it be fixed without updating the plugin? Of if not can you look through every file and confirm that it is good?

**Comment by shpp-shablyuk** [comment]
I'll take a look

**Comment by happivize** [comment]
this is urgent, please advise

**Comment by shpp-shablyuk** [comment]
The plugin for your dropbox link is not working. I downloaded the new version from https://gpldl.com and installed it on the test site. You can check how this version works. I can not verify the code of each file and give you a 100% guarantee. I checked the plugin with this service https://www.virustotal.com/en/ and with through this plugin https://wordpress.org/plugins/exploit-scanner/ and did not find anything suspicious

**Comment by happivize** [comment]
I wanted you to be able to fix it without the updated plugin is that
possible? I gave the plugin as an option but only if you can confirm all of
the code is good because they don't necessarily install viruses they
install back doors

**Comment by happivize** [comment]
can you look at this today? the main site had to be updated so now the reviews aren't showing well so need this updated

**Comment by shpp-shablyuk** [comment]
Now I don't know how long it will take me to complete this task. I can start figure it out and after a couple of hours write to you how things are going. Is this suits you?

**Comment by happivize** [comment]
Yes, please let me know after one hour, i would imagine it won't take as
long as it took to do it the first time?

**Comment by shpp-shablyuk** [comment]
I managed to make such display of reviews: https://dev.happivize.com/product/balance-your-thyroid-virginia-rounds-griffiths/#tab-reviews

**Comment by happivize** [comment]
That is looking better, the top part with the stars is missing

**Comment by shpp-shablyuk** [comment]
In the new version of the Product review plugin also don’t have this block. It seems that WooCommerce has removed this block in the new version.

**Comment by shpp-shablyuk** [comment]
Also, I must warn that I can’t be sure of the correct work of the plugin. If in the future there are any issues, then, may be,  I’ll need more hours to correct them than the cost of the plugin

**Comment by happivize** [comment]
i contacted woo about this, they said it must be an issue with flatsome, this is what they see with storefront theme with latest versions of woo and product reviews pro: http://cld.wthms.co/1JPf4j

On dev i activated the flatsome theme (deactivated child) and it still does not show. are there any snippets that could be affecting it?

the thing is i don't love this plugin, and am wondering if we need it at all if we can get what we need by customizing the default woo reviews functionality?

**Comment by shpp-shablyuk** [comment]
Do you just need to change the display of the reviews or add some additional functionality to the default woo reviews?

**Comment by happivize** [comment]
The display is the main thing, i do like that filtering by star at the top
and I like to be able to add photos.

I'm just unhappy with Skyverge lately the company that makes the plugin
we've been using, but if it's required I can continue it. I haven't heard
back from them about why the filtering by stars at the top is gone.

**Comment by happivize** [comment]
I don't understand what this means. Preview?

**Comment by shpp-shablyuk** [comment]
Sorry, I meant the reviews display

**Comment by happivize** [comment]
skyverge is saying there is likely a conflict with our theme and their
plugin and they can't guarantee it works with anything other than themes
that woocommerce sells directly. as i said they are not very helpful.

can you see if you can figure out a way maybe with another plugin to
recreate the display, ability to attach photos, and the filtering by stars?

**Comment by shpp-shablyuk** [comment]
I didn't find a plugin with this functionality. As an option, I can do it myself. Approximately this will take about 6 hours

**Comment by happivize** [comment]
OK, this will be a custom plugin then? I am good with 6 hours if you can
make it look and act like our previous version? will i able to approve and
edit the comments easily?

**Comment by shpp-shablyuk** [comment]
You can edit it via default comments edit page
https://dev.happivize.com/wp-admin/edit-comments.php

**Comment by shpp-shablyuk** [comment]
So I can Start?

**Comment by happivize** [comment]
This plugin looks like it would do most of what we need, just reformatting needed. What do you think?

https://wordpress.org/plugins/customer-reviews-woocommerce/#description

**Comment by shpp-shablyuk** [comment]
We can install it on the test site and try how it works. If it suits you, I'll customize it

**Comment by happivize** [comment]
it  looks like you have installed it, but i think the overrides you made for Product Reviews Pro are interfering?

**Comment by shpp-shablyuk** [comment]
I activated the parent theme.

**Comment by happivize** [comment]
so are you working on it?

**Comment by happivize** [comment]
Hi I responded to you on Skype on Friday, did you get that? Are you working on this? Reviews are not working right currently on live and need this to be a big priority.

**Comment by happivize** [comment]
I didn't hear back on my response on Skype - as I said this is a big priority. Are you saying you are unable to fix this? I need to know because this has been going on for sometime and I need to have it fixed.

**Comment by shpp-shablyuk** [comment]
I made that comments now show like this: https://prnt.sc/j9gwrp. You can reply to the comments through the admin panel. Does it suit you?
